MekHQ/src/mekhq/campaign/personnel/enums/PersonnelRole.java

```diff
@@ -445,19 +605,35 @@ public static PersonnelRole parseFromString(final String text) {
                 default:
                     break;
             }
-        } catch (Exception ignored) {
+        } catch (Exception ignored) {}
 
+        // <50.1 compatibility
+        switch (personnelRole) {
+            case "MECHWARRIOR" -> {
+                return MEKWARRIOR;
+            }
+            case "PROTOMECH_PILOT" -> {
+                return PROTOMEK_PILOT;
+            }
+            case "MECH_TECH" -> {
+                return MEK_TECH;
+            }
+            case "AERO_TECH" -> {
+                return AERO_TEK;
+            }
+            default -> {}
         }
 
+        // Error report, if parsing fails.
         MMLogger.create(PersonnelRole.class)
-                .error("Unable to parse " + text + " into a PersonnelRole. Returning NONE.");
+                .error("Unable to parse {} into a PersonnelRole. Returning NONE.", personnelRole);
```

Comment:
The MMLogger doesn't actually have a method that will do this. Will need to use a String.format() otherwise it'll pop up a dialog.

Reply:
Oh that is _really_ good to know. I'll pop in a correction real quick.

Is there any way to add this functionality to MMLogger, because IDEA recommends doing what I did here and it's going to be a bit of a pain having to remember to check devs aren't following IDEA's recommendation whenever we review PRs

Reply:
Would require a re-write of those MMLogger methods to go from `error` to `errorWithDialog` (and related) which isn't a bad idea over all. It would allow mimicking the Log4J commands but would require a re-write in the few places that actually use the current method for a dialog first.

Reply:
Thanks for the heads up, btw. I was at a loss why I was getting a dialog box whenever I ran the personnel tests, but had assumed it was because I was testing in a non-headless environment.

Should be fixed now :)